djpsa/halo/views.py

```diff
@@ -0,0 +1,45 @@
+import json
+import logging
+
+from braces import views
+from django.views.generic import View
+from django.http import HttpResponse
+
+from djpsa.halo.records.ticket.sync import TicketSynchronizer
+
+logger = logging.getLogger(__name__)
+
+
+class CallBackView(views.CsrfExemptMixin,
+                   views.JsonRequestResponseMixin, View):
+
+    CALLBACK_TYPES = {
+        'ticket': TicketSynchronizer,
+    }
+
+    def post(self, request, *args, **kwargs):
+        logger.debug(f"Received callback {request.path} {request.method}")
+
+        try:
+            data = json.loads(request.body.decode('utf-8'))
+        except json.JSONDecodeError as e:
+            logger.error('Error decoding JSON for callback: %s', e)
+            return HttpResponse(status=400)
+
+        sync = None
+
+        for record_type, sync_class in self.CALLBACK_TYPES.items():
+            # Searches for record type keys in data, i.e. if it's a ticket
+            # callback, it will find 'ticket'.
+            if record_type in data:
```

Comment:
This seems fragile. What if/when we have a project callback, but the ticket callback includes the field 'project'? Then we could use the project synchronizer for ticket callbacks.

Can we not have different URLs for different record types? Assuming we can, then we should offer a urls.py module with one URL for each record type (since each type has its own view), and then the application can just `include` them at whatever URL the developer chooses.

Reply:
The dict we get back is the webhook istelf, so its: "Webhook_id", "event", "notification_id", etc, and then "ticket" which is the object we care about, so that problem can't happen. The plan is to offer different urls for different types though (as discussed in slack) if we ever implement them, but unfortunately ticket and project will always come through the same url.